crates/sallyport/src/guest/handler.rs

```diff
@@ -254,6 +267,54 @@ pub trait Handler {
         self.execute(syscall::Fstat { fd, statbuf })?
     }
 
+    /// Executes [`futex`](https://man7.org/linux/man-pages/man2/futex.2.html) syscall.
+    fn futex(
+        &mut self,
+        uaddr: &mut AtomicU32,
+        futex_op: c_int,
+        val: u32,
+        timespec: Option<&timespec>,
+        _uaddr2: Option<&mut AtomicU32>,
+        val3: u32,
+    ) -> Result<c_long> {
+        let futex_op = futex_op & !FUTEX_PRIVATE_FLAG;
+
+        match futex_op {
+            FUTEX_WAIT => {
+                let timeout = timespec.map(|timespec| ParkTimeout {
+                    timespec: *timespec,
+                    absolute: false,
+                });
+
+                while uaddr.load(Ordering::Relaxed) == val {
+                    let _ = self.park(timeout.as_ref());
+                }
+                Ok(0)
+            }
+            FUTEX_WAIT_BITSET => {
+                if val3 != !0u32 {
+                    return Err(ENOTSUP);
+                }
+
+                let timeout = timespec.map(|timespec| ParkTimeout {
+                    timespec: *timespec,
+                    absolute: true,
+                });
+
+                while uaddr.load(Ordering::Relaxed) == val {
+                    let _ = self.park(timeout.as_ref());
+                }
+                Ok(0)
+            }
+            FUTEX_WAKE => {
+                // TODO: return the number of woken threads
```

Comment:
Can we file an issue about it, such that we don't forget about this?

Reply:
added a comment